pkg/core/resources/apis/mesh/validators.go

```diff
@@ -151,6 +151,36 @@ func ValidatePort(path validators.PathBuilder, port uint32) validators.Validatio
 	return err
 }
 
+const dnsLabel = `[a-z0-9]([-a-z0-9]*[a-z0-9])?`
+
+var domainRegexp = regexp.MustCompile("^" + dnsLabel + "(\\." + dnsLabel + ")*" + "$")
+
+// ValidateHostname validates a gateway hostname field. A hostname may be one of
+//	- '*'
+//	- '*.domain.name'
+//	- 'domain.name'
+func ValidateHostname(path validators.PathBuilder, hostname string) validators.ValidationError {
+	err := validators.ValidationError{}
+
+	if hostname == "*" {
+		return err
```

Comment:
Can we return `nil` here? I was thinking there is a controversy between this line and the comment above :)

Reply:
The common pattern here is to return an empty validation error on success, so this seems ok?

Reply:
yeah, that's true, maybe `return validators.ValidationError{}` then?

Reply:
There's no functional difference with that, but if it's the preferred style I could do that? The other validation helpers used named error return, which is doing the same thing but harder to read :joy:

Reply:
That's not for the functional difference, when I read it for the first time I was 100 percent sure we don't support `"*"` :D
```go
if hostname == "*" {
    return err
}
```

Reply:
kk, I made the suggested change. I have an idea to make this read more clearly in a followup PR.